src/credentials/FabricTable.cpp

```diff
@@ -214,26 +226,29 @@ CHIP_ERROR FabricInfo::DeleteFromStorage(PersistentStorageDelegate * storage, Fa
 {
     CHIP_ERROR err = CHIP_NO_ERROR;
 
-    char key[kKeySize];
-    ReturnErrorOnFailure(GenerateKey(fabricIndex, key, sizeof(key)));
+    DefaultStorageKeyAllocator keyAlloc;
 
-    err = storage->SyncDeleteKeyValue(key);
+    // Try to delete all the state even if one of the deletes fails.
+    typedef const char * (DefaultStorageKeyAllocator::*KeyGetter)(FabricIndex);
+    constexpr KeyGetter keyGetters[] = { &DefaultStorageKeyAllocator::FabricNOC, &DefaultStorageKeyAllocator::FabricICAC,
+                                         &DefaultStorageKeyAllocator::FabricRCAC, &DefaultStorageKeyAllocator::FabricMetadata,
+                                         &DefaultStorageKeyAllocator::FabricOpKey };
+    for (auto & keyGetter : keyGetters)
+    {
+        CHIP_ERROR getterErr = storage->SyncDeleteKeyValue((keyAlloc.*keyGetter)(fabricIndex));
+        // Keys not existing is not really an error condition.
+        if (err == CHIP_NO_ERROR && getterErr != CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND)
```

Comment:
Seems like a mistake when read here, but it's because `err` is used as a `previousErr`. Suggest making a new `CHIP_ERROR prevGetterErr = CHIP_NO_ERROR` at first and doing `prevGetterErr = getteErr` in this if.

Reply:
Done.